File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        left = 0
        right = len(height) - 1
        maxWater = 0

        while left < right:
            width = right - left
            water = min(height[left], height[right]) * width

            maxWater = max(maxWater, water)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return maxWater

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxWater = 0;

        while (left < right) {
            int width = right - left;

            int minHeight = Math.min(height[left], height[right]);

            int water = minHeight * width;

            maxWater = Math.max(maxWater, water);

            if (height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxWater;
    }
}